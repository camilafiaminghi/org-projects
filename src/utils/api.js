import axios from 'axios'

const RESTAPI_URL = 'https://api.github.com'
const RESTAPI_AUTH = 'token dd0f9fb05e1398f0784dd208d44d141b46134d5f'

// https://api.github.com/search/repositories?q=org:facebook+sort:stars&per_page=3
// https://api.github.com/repos/facebook/react/branches

export const getReposByOrg = (org, sort='starts') => {
	return axios.get(`${RESTAPI_URL}/search/repositories?q=org:${org}+sort:stars&per_page=4`, {
		headers: {
			'Accept': 'application/json',
			'Authorization': RESTAPI_AUTH
		}
	})
	.then(response => response.data)
	.catch(error => { throw error })
}

export const getBranches = (action) => {
	return axios.get(`${RESTAPI_URL}/repos/${action.org}/${action.repo}/branches`, {
		headers: {
			'Accept': 'application/json',
			'Authorization': RESTAPI_AUTH
		}
	})
	.then(response => response.data)
	.catch(error => { throw error })
}
