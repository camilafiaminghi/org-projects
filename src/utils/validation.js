export function validationRules (name, value) {
	const rule = {
		search: !!value.trim() && value.length > 3
	}
	return rule[name]
}

export function isValid (form) {
	return Object.keys(form).every(function (key) {
		return form[key]
	});
}
