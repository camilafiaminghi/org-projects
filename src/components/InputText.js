import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { validationRules } from '../utils/validation'

class InputText extends Component {
	static propTypes = {
		name: PropTypes.string.isRequired,
		placeholder: PropTypes.string.isRequired,
		maxLength: PropTypes.number.isRequired,
		handleOnChange: PropTypes.func.isRequired,
		submitted: PropTypes.bool,
		charsLeft: PropTypes.bool,
		message: PropTypes.string,
		value: PropTypes.string
	}

	state = {
		text: '',
		changed: false,
		valid: false
	}

	handleValidation = (event) => {
		const value = event.target.value

		if ( value.length > 0 ) {
			const { name } = this.props
			const valid = validationRules(name, value)

			this.setState((prevState) => ({
				...prevState,
				text: value,
				changed: true,
				valid
			}))

			this.props.handleOnChange(name, value, valid)
		}
	}

	render() {
		const { changed, valid, text } = this.state
		const { name, placeholder, maxLength, charsLeft, message, submitted } = this.props
		const textLeft = maxLength - text.length

		return (
			<div>
				<input
					type="text"
					name={name}
					placeholder={placeholder}
					maxLength={maxLength}
					value={text}
					onChange={this.handleValidation}
					autoComplete="off" />

				{ (textLeft <= (maxLength - Math.round(maxLength/2)) && charsLeft && changed) && (<span>characteres left [{ textLeft }]</span>) }
				{ ((!valid && changed) && submitted) && (<span>{message}</span>) }
			</div>
		)
	}
}

export default InputText
