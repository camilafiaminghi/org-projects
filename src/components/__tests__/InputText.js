import React from 'react'
import renderer from 'react-test-renderer'
import InputText from './../InputText'
import { validation, isValid } from './../../utils/validation'

let props
let wrapper
const state = {
	text: '',
	changed: false,
	valid: false
}

describe('<InputText />', () => {

	beforeEach(() => {
		props = {
			name: 'search',
			placeholder: '',
			maxLength: 122,
			handleOnChange: jest.fn(),
			submitted: false,
			charsLeft: false,
			message: '',
			value: ''
		}
		wrapper = shallow(<InputText {...props} />)
	})

	it('matches snapshot', () => {
		const component = renderer.create(<InputText {...props} />)
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render', () => {
		expect(wrapper.exists()).toBeTruthy()
	})

	it('should handleValidation', () => {
		const input = wrapper.find('input')
		input.simulate('change', {target: {value: 'test'}})

		expect(wrapper.state('text')).toEqual('test')
		expect(props.handleOnChange).toHaveBeenCalledTimes(1)
	})
})
