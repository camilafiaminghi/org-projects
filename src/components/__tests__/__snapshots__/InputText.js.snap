// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText /> matches snapshot 1`] = `
<input
  autoComplete="off"
  maxLength={122}
  name="search"
  onChange={[Function]}
  placeholder=""
  type="text"
  value=""
/>
`;
